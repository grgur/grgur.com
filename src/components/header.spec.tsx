import React from "react"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"
import Header from "./header"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders the title", () => {
    const { getByTestId } = render(<Header />)
    expect(getByTestId("title")).toHaveTextContent("GrgurdeGrisogono")
  })
})
