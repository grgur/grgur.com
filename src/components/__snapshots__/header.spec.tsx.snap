// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="relative overflow-hidden bg-gray-50"
>
  <div
    className="hidden bg-fixed bg-right bg-no-repeat bg-contain sm:block sm:absolute sm:inset-y-0 sm:h-full sm:w-full sm:bg-local md:bg-scroll lg:bg-local xl:bg-fixed"
    style={
      Object {
        "backgroundColor": "#0D0403",
        "backgroundImage": "url(https://res.cloudinary.com/grguru/image/upload/c_scale,e_art:incognito,f_auto,w_1024/v1593166618/grgur/J1920x1080-00355.jpg)",
      }
    }
  />
  <div
    className="relative pt-6 pb-12 sm:pb-16 md:pb-20 lg:pb-28 xl:pb-32"
  >
    <div
      className="max-w-screen-xl px-4 mx-auto sm:px-6"
    >
      <nav
        className="relative flex items-center justify-between sm:h-10 md:justify-center"
      >
        <div
          className="flex items-center flex-1 md:absolute md:inset-y-0 md:left-0"
        >
          <div
            className="flex items-center justify-between w-full md:w-auto"
          >
            <a
              aria-label="Home"
              href="#"
            />
            <div
              className="flex items-center -mr-2 md:hidden"
            >
              <button
                aria-haspopup="true"
                aria-label="Main menu"
                className="inline-flex items-center justify-center p-2 text-gray-400 transition duration-150 ease-in-out rounded-md hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 focus:text-gray-500"
                id="main-menu"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="w-6 h-6"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 6h16M4 12h16M4 18h16"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </nav>
    </div>
    <main
      className="max-w-screen-xl px-4 mx-auto mt-10 sm:mt-12 sm:px-6 md:mt-16 lg:mt-20 xl:mt-28"
    >
      <div
        className="text-center"
      >
        <h2
          className="text-4xl font-extrabold leading-10 tracking-tight text-gray-900 sm:text-5xl sm:leading-none md:text-6xl"
        >
          Grgur
          <br
            className="xl:hidden"
          />
          <span
            className="xl:ml-2"
          />
          <span
            className="text-indigo-600"
          >
            de
          </span>
          <span
            className="xl:ml-2"
          />
          Grisogono
        </h2>
        <p
          className="max-w-md mx-auto mt-3 text-base text-gray-500 sm:text-lg md:mt-5 md:text-xl md:max-w-3xl"
        >
          You've likely interacted with a product I helped develop, whether in the automotive, financial, food, or retail industry. Hopefully, you enjoyed it, and I was successful in taking customer experience and performance to the next level. I don't like to be called a team lead or a principal architect because I feel equal to anyone else on the team. During my adventures with many Fortune 500 companies, startups, and governments, I learned the value of being a positive influence that extracts the best of everyone on the team, instead of being a rockstar that draws all the spotlight. You'll probably recognize that web technologies are my strong suit. I love contributing to open source projects of any size, and I'm particularly proud of my involvement with Vue, Ionic, React Router, and Webpack. However, I feel that my continuous learning about human psychology is one of my greatest assets. With my B.S. degree in Service Leadership, I learned how to apply basic science of how our brains work to ensure better service delivery, teamwork, user experience, performance, conversion, negotiation, etc. If I'm lucky to work with you, you'll notice that I'm easy to be around and focus on our mutual success. I'll try to coach, but I won't get in the way. I'll try to reevaluate everything we do to ensure we are doing the right thing. I'll ask questions instead of making statements. I'll do my best to help you feel comfortable with flops. On top of it, I'll find any opportunities to learn from you. I doubt that you care about my many years of experience or that I organized conferences or co-authored a book or even the certifications I hold. While that's all nice, I'm actually proud to be a husband to a wonderful wife and a dad to a couple of amazing kids. They push me to get better at everything I do. Feel free to connect with me, drop me a message, ask a question.
        </p>
      </div>
    </main>
  </div>
</header>
`;
